src/db_api_util.ts

```diff
@@ -86,6 +86,184 @@ export type DatabaseCommandEntry = {
+  slackParentMessageTs: string
+): Promise<string | null> {
+  const client = await pool.connect();
+  try {
+    const result = await client.query(
+      `SELECT slack_message_ts FROM messages
+      WHERE slack_parent_message_ts=$1 AND slack_message_ts IS NOT NULL
+      ORDER BY COALESCE(slack_send_timestamp, slack_receive_timestamp) DESC LIMIT 1`,
+      [slackParentMessageTs]
+    );
+    if (result.rows.length > 0) {
+      return result.rows[0]['slack_message_ts'];
+    }
+    return null;
+  } finally {
+    client.release();
+  }
+}
+
+export async function getUnclaimedVoters(
+  channelId: string | null
+): Promise<ThreadInfo[]> {
+  const client = await pool.connect();
+  try {
+    let result = null;
+    if (channelId) {
+      result = await client.query(
+        `SELECT
+         t.slack_parent_message_ts as id, t.channel_id, t.user_phone_number, t.user_id, EXTRACT(EPOCH FROM now() - t.updated_at) as age
+        FROM threads t
+        WHERE t.needs_attention AND t.channel_id = $1
+          AND NOT EXISTS (SELECT FROM volunteer_voter_claims c WHERE t.user_id=c.user_id
+          AND t.user_phone_number=c.user_phone_number)`,
+        [channelId]
+      );
+    } else {
+      result = await client.query(
+        `SELECT
+         t.slack_parent_message_ts as id, t.channel_id, t.user_phone_number, t.user_id, EXTRACT(EPOCH FROM now() - t.updated_at) as age
+        FROM threads t
+        WHERE t.needs_attention AND NOT EXISTS (SELECT FROM volunteer_voter_claims c WHERE t.user_id=c.user_id
+          AND t.user_phone_number=c.user_phone_number)`
+      );
+    }
+    return result.rows.map((x) => ({
+      slackParentMessageTs: x['id'],
+      channelId: x['channel_id'],
+      userPhoneNumber: x['user_phone_number'],
+      userId: x['user_id'],
+      age: x['age'],
+    }));
+  } catch (error) {
+    logger.info('Failed to query unclaimed threads; ignoring for now!');
+    return [];
+  } finally {
+    client.release();
+  }
+}
+
+export async function getThreadsNeedingAttentionFor(
+  slackUserId: string
+): Promise<ThreadInfo[]> {
+  const client = await pool.connect();
+  try {
+    const result = await client.query(
+      `SELECT
+         t.slack_parent_message_ts as id, t.channel_id, t.user_phone_number, t.user_id, EXTRACT(EPOCH FROM now() - t.updated_at) as age
+        FROM threads t, volunteer_voter_claims c
+        WHERE t.needs_attention
+          AND t.user_id=c.user_id
+          AND t.user_phone_number=c.user_phone_number
+          AND c.volunteer_slack_user_id=$1`,
+      [slackUserId]
+    );
+    return result.rows.map((x) => ({
+      slackParentMessageTs: x['id'],
+      channelId: x['channel_id'],
+      userPhoneNumber: x['user_phone_number'],
+      userId: x['user_id'],
+      age: x['age'],
+    }));
+  } catch (error) {
+    logger.info('Failed to query threads; ignoring for now!');
+    return [];
+  } finally {
+    client.release();
+  }
+}
+
+export async function getThreadNeedsAttentionFor(
```

Comment:
I see the difference between getThreadNeedsAttentionFor and getThreadsNeedingAttentionFor but can you please add comments since it can be confusing?